numpy/testing/_private/utils.py

```diff
@@ -876,6 +880,9 @@ def assert_array_equal(x, y, err_msg='', verbose=True):
         The error message to be printed in case of failure.
     verbose : bool, optional
         If True, the conflicting values are appended to the error message.
+    strict : bool, optional
+        If True, raise an assertion when one of the array_like objects is a
+        scalar or if `x` and `y` have a different data type.
```

Comment:
This isn't quite right.  An error is not raised if *both* inputs are scalar (as long as the data types are the same).  Maybe a better way to phrase this is `"If True, raise an exception when the shapes of the inputs are not the same, or ..."`.

Reply:
I get you, but the previous behaviour would also error when the shapes were not the same. Maybe something like "raise an assertion when only one of..." or "just one" would do it? I was trying to avoid a longer sentence like "when one ... is a scalar and one is not".

Reply:
Maybe something like “disables the broadcasting of scalars, thus matching the behavior used for all other shapes”?

Reply:
@jontwo, the updated text looks good, thanks.